### ISSUE TITLE
[BUG] arena_memory_resource allows creation with a maximum allocation size smaller than its minimum superblock size

### ISSUE THREAD

**Issue by harrism** [rapidsai/rmm#919]
**Describe the bug**
Creating an `arena_memory_resource` with a value of the constructor parameter `maximum_size` smaller than its internal minimum superblock size will cause any allocation to fail. 

Also, the default values of the parameters `initial_size` and `maximum_size` are defined in the detail namespace. Since there is another defaulted parameter (`dump_log_on_failure`) after these two parameters, this makes it difficult to construct an instance of this MR using the defaults for the sizes but a non-default value for `dump_log_on_failure`.

**Steps/Code to reproduce bug**

```
cuda_mr cuda;
arena_mr mr{&cuda, 1024_B, 32_KiB, true};
mr.allocate(1024_B); // <-- exception
```

**Expected behavior**
An exception should be thrown by the constructor if invalid parameters are passed.

**Comment by harrism** [comment]
@rongou is this fixed in #916 as well?

**Comment by rongou** [comment]
Added a check and corresponding test.
